### PR TITLE
Fix Menu

### PR DESCRIPTION
**Pull request by sharix4**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/components/PrivateLayout/useOneClickOutside.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+function useOnClickOutside<T extends HTMLElement = HTMLElement>(
+  ref: RefObject<T>,
+  handler: (event: AnyEvent) => void,
+) {
+  useEffect(() => {
+    const listener = (event: AnyEvent) => {
+      const el = ref?.current
+
+      // Do nothing if clicking ref's element or descendent elements
+      if (!el || el.contains(event.target as Node)) {
```
Comment (danylo-melnychuk) [inline src/components/PrivateLayout/useOneClickOutside.tsx:14]:
```suggestion
      if (!ref.current || el.contains(event.target as Node)) {
```

File: src/components/PrivateLayout/useOneClickOutside.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+type AnyEvent = MouseEvent | TouchEvent
+
+function useOnClickOutside<T extends HTMLElement = HTMLElement>(
+  ref: RefObject<T>,
+  handler: (event: AnyEvent) => void,
+) {
+  useEffect(() => {
+    const listener = (event: AnyEvent) => {
+      const el = ref?.current
```
Comment (danylo-melnychuk) [inline src/components/PrivateLayout/useOneClickOutside.tsx:11]: We don't need this variable

**Review by danylo-melnychuk — APPROVED** [review]
Briliant

File: src/components/PrivateLayout/useOneClickOutside.tsx (hunk 2 above)
Comment (sharix4) [reply src/components/PrivateLayout/useOneClickOutside.tsx:11]: you're right but we don't need this variable
  if (!ref?.current|| ref?.current.contains(event.target as Node)) {

File: src/components/PrivateLayout/useOneClickOutside.tsx (hunk 1 above)
Comment (sharix4) [reply src/components/PrivateLayout/useOneClickOutside.tsx]: if (!ref?.current|| ref?.current.contains(event.target as Node)) {

**Review by christykobyletska — APPROVED** [review]

**Review by Vinnytile — APPROVED** [review]

**Pull request merged**